src/jit/importer.cpp

```diff
@@ -12485,21 +12485,36 @@ void Compiler::impImportBlockCode(BasicBlock* block)
 
                 type = genActualType(lclTyp);
 
-#if SMALL_TREE_NODES
-                if (callNode)
+                // If this is a no-op cast, just use op1.
+                if (!ovfl && (type == op1->TypeGet()) && (genTypeSize(type) == genTypeSize(lclTyp)))
                 {
-                    op1 = gtNewCastNodeL(type, op1, uns, lclTyp);
+                    // Nothing needs to change
                 }
-                else
-#endif // SMALL_TREE_NODES
+                else if (!ovfl && (lclTyp == op1->TypeGet()) && (genTypeSize(type) == genTypeSize(lclTyp)))
                 {
-                    op1 = gtNewCastNode(type, op1, uns, lclTyp);
+                    // Bash operand to right type
+                    op1->gtType = lclTyp;
```

Comment:
I don't follow - this code is executed when `lclType == op1->gtType` so the assignment is pointless.

Reply:
Oops, thanks. I should be bashing it to `type` here (the operand stack type).

Reply:
And probably this second case never happens since operands should not have non-stack types.

Reply:
Yes, it's probably better to remove this. The code is somewhat difficult to understand as it is.